### PR TITLE
Including optional time for isValid in server

### PR DESCRIPTION
**Pull request by HughParry**

### CODE REVIEW

**Review by forgetso — CHANGES_REQUESTED** [review]
Looks good, could you stick the time stuff in a separate function?

File: packages/types/src/provider/api.ts [hunk 1]
```diff
@@ -68,6 +69,7 @@ export type CaptchaSolutionBodyType = zInfer<typeof CaptchaSolutionBody>
 export const VerifySolutionBody = object({
     [ApiParams.user]: string(),
     [ApiParams.commitmentId]: string().optional(),
+    [ApiParams.maxVerifiedTime]: number().optional(),
```
Comment (goastler) [inline packages/types/src/provider/api.ts:78]: is this something the client could fudge?

File: packages/provider/src/tasks/tasks.ts [hunk 2]
```diff
@@ -541,4 +541,19 @@ export class Tasks {
...
+     * Get the current block number
+     */
+    async getCurrentBlockNumber(): Promise<number> {
+        return (await getBlockNumber(this.contract.api)).toNumber()
+    }
+
+    /**
+     * Get the current block time in milliseconds
+     */
+    async getBlockTimeMs(): Promise<number> {
```
Comment (goastler) [inline packages/provider/src/tasks/tasks.ts]: this should be a fn on the contract api, no?

File: packages/provider/src/tasks/tasks.ts [hunk 3]
```diff
@@ -541,4 +541,19 @@ export class Tasks {
     async getProviderDataset(datasetId: string): Promise<DatasetWithIds> {
         return await this.db.getDataset(datasetId)
     }
+
+    /**
+     * Get the current block number
+     */
+    async getCurrentBlockNumber(): Promise<number> {
```
Comment (goastler) [inline packages/provider/src/tasks/tasks.ts]: this should be a fn on the contract api, no?

**Review by goastler — DISMISSED** [review]

**Review by forgetso — APPROVED** [review]

**Review by forgetso — APPROVED** [review]

**Pull request merged**